From chrisw@sous-sol.org Thu Apr 17 02:29:12 2008
X-Envelope-From: <stable-commits-owner@vger.kernel.org>
X-Envelope-To: <vger@sl-h.de>
X-Delivery-Time: 1208392206
X-UID: 208984
Return-Path: <stable-commits-owner@vger.kernel.org>
X-RZG-CLASS-ID: mi
Received: from vger.kernel.org ([209.132.176.167])
	by mailin.webmailer.de (ahmed-mailin mi29) (RZmta 16.24)
	with ESMTP id x06a66k3H09Mjt for <vger@sl-h.de>;
	Thu, 17 Apr 2008 02:30:06 +0200 (MEST)
	(envelope-from: <stable-commits-owner@vger.kernel.org>)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1753794AbYDQAaE (ORCPT <rfc822;vger@sl-h.de>);
	Wed, 16 Apr 2008 20:30:04 -0400
Received: (majordomo@vger.kernel.org) by vger.kernel.org id S1753712AbYDQAaE
	(ORCPT <rfc822;stable-commits-outgoing>);
	Wed, 16 Apr 2008 20:30:04 -0400
Received: from sous-sol.org ([216.99.217.87]:53594 "EHLO sous-sol.org"
	rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
	id S1753585AbYDQAaD (ORCPT <rfc822;stable-commits@vger.kernel.org>);
	Wed, 16 Apr 2008 20:30:03 -0400
Received: from localhost.localdomain (localhost.localdomain [127.0.0.1])
	by sous-sol.org (8.14.2/8.14.2) with ESMTP id m3H0TCpt005094;
	Wed, 16 Apr 2008 17:29:12 -0700
Message-Id: <200804170029.m3H0TCpt005094@sous-sol.org>
Subject: patch sunrpc-fix-a-memory-leak-in-rpc_create.patch queued to 2.6.24-stable tree
To: Trond.Myklebust@netapp.com,
 chrisw@sous-sol.org,
 chuck.lever@oracle.com
Cc: stable@kernel.org,
 stable-commits@vger.kernel.org
From:	chrisw@sous-sol.org
Date:	Wed, 16 Apr 2008 17:29:12 -0700
In-Reply-To: <20080409011532.22409.87741.stgit@c-69-242-210-120.hsd1.mi.comcast.net>
X-Virus-Scanned: ClamAV version 0.93rc1, clamav-milter version devel-20080219 on sequoia.sous-sol.org
X-Virus-Status:	Clean
X-Spam-Status: No, score=-4.4 required=5.0 tests=ALL_TRUSTED,AWL,BAYES_00,
	MISSING_MID autolearn=ham version=3.2.3
X-Spam-Checker-Version:	SpamAssassin 3.2.3 (2007-08-08) on sequoia.sous-sol.org
Sender:	stable-commits-owner@vger.kernel.org
Precedence: bulk
Reply-To: linux-kernel@vger.kernel.org
List-ID: <stable-commits.vger.kernel.org>
X-Mailing-List:	stable-commits@vger.kernel.org
Status: R
X-Status: NC
X-KMail-EncryptionState:  
X-KMail-SignatureState:  
X-KMail-MDN-Sent:  


This is a note to let you know that we have just queued up the patch titled

     Subject: SUNRPC: Fix a memory leak in rpc_create()

to the 2.6.24-stable tree.  Its filename is

     sunrpc-fix-a-memory-leak-in-rpc_create.patch

A git repo of this tree can be found at 
    http://www.kernel.org/git/?p=linux/kernel/git/stable/stable-queue.git;a=summary


>From stable-bounces@linux.kernel.org  Wed Apr 16 16:27:25 2008
From: Trond Myklebust <Trond.Myklebust@netapp.com>
To: stable@kernel.org
Date: Tue, 08 Apr 2008 21:15:33 -0400
Message-ID: <20080409011532.22409.87741.stgit@c-69-242-210-120.hsd1.mi.comcast.net>
Cc: Chuck Lever <chuck.lever@oracle.com>
Subject: SUNRPC: Fix a memory leak in rpc_create()

From: Chuck Lever <chuck.lever@oracle.com>

upstream commit: ed13c27e546667fb0967ae30f5070cd7f6455f90

Commit 510deb0d was supposed to move the xprt_create_transport() call in
rpc_create(), but neglected to remove the old call site.  This resulted in
a transport leak after every rpc_create() call.

This leak is present in 2.6.24 and 2.6.25.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
---

 net/sunrpc/clnt.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/net/sunrpc/clnt.c
+++ b/net/sunrpc/clnt.c
@@ -249,10 +249,6 @@ struct rpc_clnt *rpc_create(struct rpc_c
 	};
 	char servername[20];
 
-	xprt = xprt_create_transport(&xprtargs);
-	if (IS_ERR(xprt))
-		return (struct rpc_clnt *)xprt;
-
 	/*
 	 * If the caller chooses not to specify a hostname, whip
 	 * up a string representation of the passed-in address.


Patches currently in stable-queue which might be from Trond.Myklebust@netapp.com are

queue-2.6.24/sunrpc-fix-a-memory-leak-in-rpc_create.patch
--
To unsubscribe from this list: send the line "unsubscribe stable-commits" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

